### PR TITLE
Create entities collection

### PR DESCRIPTION
**Pull request by pdcp1**
### This PR:

1. Creates a new `entities` collection with fields:
```
{
  "_id": ObjectId("6357263070b2d2aaca6c7093"),
  "id": "boston-university",
  "name": "Boston University"
}
```
2. Updates all incidents to use the Entity `id` fields instead of Entity name on these fields:
  2.1 `Alleged deployer of AI system`
  2.2 `Alleged developer of AI system`
  2.3 `Alleged harmed or nearly harmed parties`

ie: From `Boston University` to `boston-university`

3. Updates all submissions to use the Entity `id` fields instead of Entity name on these fields:
  3.1 `deployers`
  3.2 `developers`
  3.3 `harmed_parties`

ie: From `Boston University` to `boston-university`

4. Update `createEntitiesPages.js` page creator to use this new structure

### Test

This is the link to test this PR: https://deploy-preview-10--pablocosta-staging.netlify.app/

### CODE REVIEW

**Review by cesarvarela — COMMENTED** [review]
2 questions:

- When will we handle when new entities are added?
- With this change editors will have to use entity ids now?

**Comment by pdcp1** [comment]
@cesarvarela 
> * When will we handle when new entities are added?

That is not covered in this PR. My suggestion is to change the input for these 3 Incident fields to have a text suggestion for the existing Entities, and when the user creates/updates the Incident, save the new entities on the DB and use the entity ID.

> * With this change editors will have to use entity ids?

No, Editors can use the same Entity name text as before.

**Comment by pdcp1** [comment]
@cesarvarela I added the logic to handle new Entities.
Basically, when a submission or an Incident is created or edited, we check the Entities associated and if there are new ones, we create them on the `entities` collection and use the Entity ID just generated.

This PR is ready for review. The link to test is https://deploy-preview-10--pablocosta-staging.netlify.app

File: site/gatsby-site/src/pages/incidents/edit.js [hunk 1]
```diff
@@ -6,24 +6,42 @@ import useToastContext, { SEVERITY } from '../../hooks/useToast';
...
 
   const [incidentId] = useQueryParam('incident_id', withDefault(NumberParam, 1));
 
   const { data: incidentData, loading } = useQuery(FIND_INCIDENT, {
     variables: { query: { incident_id: incidentId } },
   });
 
+  const {
+    entities: { nodes: allEntities },
+  } = useStaticQuery(graphql`
```
Comment (cesarvarela) [inline site/gatsby-site/src/pages/incidents/edit.js:30]: This will lag behind what is in the actual DB, is it better to query the Atlas Graphql instead? 🤔

**Review by cesarvarela — APPROVED** [review]
A small comment that can be fixe in another PR

File: site/gatsby-site/src/pages/incidents/edit.js (hunk 1 above)
Comment (pdcp1) [reply site/gatsby-site/src/pages/incidents/edit.js:30]: Issue created https://github.com/responsible-ai-collaborative/aiid/issues/1354

**Pull request merged**